src/Psalm/Internal/Analyzer/Statements/Expression/SimpleTypeInferer.php

```diff
@@ -55,7 +55,8 @@ public static function infer(
         Aliases $aliases,
         FileSource $file_source = null,
         ?array $existing_class_constants = null,
-        ?string $fq_classlike_name = null
+        ?string $fq_classlike_name = null,
+        bool $const_inference = false
```

Comment:
why did you need to add a flag here?

Reply:
I mistakenly thought that adding `New_` would break the existing inference.